nucypher/blockchain/eth/sol/source/contracts/staking_contracts/StakingInterface.sol

```diff
@@ -5,62 +5,99 @@ import "contracts/staking_contracts/AbstractStakingContract.sol";
 import "contracts/NuCypherToken.sol";
 import "contracts/StakingEscrow.sol";
 import "contracts/PolicyManager.sol";
+import "contracts/WorkLock.sol";
 
 
 /**
-* @notice Interface for accessing main contracts from a staking contract
-* @dev All methods must be stateless because this code will be executed by delegatecall call.
-* If state is needed - use getStateContract() method to access state of this contract.
-* @dev |v1.3.1|
+* @notice Base StakingInterface
 */
-contract StakingInterface {
-
-    event DepositedAsStaker(address indexed sender, uint256 value, uint16 periods);
-    event WithdrawnAsStaker(address indexed sender, uint256 value);
-    event Locked(address indexed sender, uint256 value, uint16 periods);
-    event Divided(address indexed sender, uint256 index, uint256 newValue, uint16 periods);
-    event Mined(address indexed sender);
-    event PolicyRewardWithdrawn(address indexed sender, uint256 value);
-    event MinRewardRateSet(address indexed sender, uint256 value);
-    event ReStakeSet(address indexed sender, bool reStake);
-    event ReStakeLocked(address indexed sender, uint16 lockUntilPeriod);
-    event WorkerSet(address indexed sender, address worker);
-    event Prolonged(address indexed sender, uint256 index, uint16 periods);
-    event WindDownSet(address indexed sender, bool windDown);
+contract BaseStakingInterface {
 
     NuCypherToken public token;
     StakingEscrow public escrow;
     PolicyManager public policyManager;
+    WorkLock public workLock;
 
     /**
     * @notice Constructor sets addresses of the contracts
     * @param _token Token contract
     * @param _escrow Escrow contract
     * @param _policyManager PolicyManager contract
+    * @param _workLock WorkLock contract
     */
     constructor(
         NuCypherToken _token,
         StakingEscrow _escrow,
-        PolicyManager _policyManager
+        PolicyManager _policyManager,
+        WorkLock _workLock
     )
         public
     {
         require(_token.totalSupply() > 0 &&
             _escrow.secondsPerPeriod() > 0 &&
-            _policyManager.secondsPerPeriod() > 0);
+            _policyManager.secondsPerPeriod() > 0 &&
+            // in case there is no worklock contract
+            (address(_workLock) == address(0) || _workLock.startBidDate() > 0));
```

Comment:
Does `no worklock` in this case describe the situation where contracts are deployed (but not initialized) before WorkLock is deployed?

Reply:
It's more for situation when we deploy only main contracts (for example, some testnet)